Marlin/Marlin_main.cpp

```diff
@@ -4053,6 +4054,13 @@ inline void gcode_M109() {
 #endif // HAS_TEMP_BED
 
 /**
+ * M110: Set Current Line Number
+ */
+inline void gcode_M110() {
+  if (code_seen('N')) gcode_N = code_value_long();
```

Comment:
@thinkyhead 
Take an other look on when `process_next_command()` is called and when not.